### PR TITLE
Week 2 paul gooden

### PR DESCRIPTION
**Pull request by Phalco1**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [ X] Coding Concepts
- [ X] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

I didn't do the last question on the coding challenges due to my wife being sick and I have 3 children. Plus I had 2 VA appointments that drew my attention away so I had to make up with studying. But I did the best I could and I think that should be the only unanswered question in the assessment beside the stretch questions.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -1,55 +1,111 @@
...
+
+// // // 4.) Create a function with a conditional statement if the object is divisible by 3 then itll return that number is divisible by 3. if else, itll return the number is not divisible by 3.
+
+// // // 1.) CREATE
+
+// describe("divideBy3", () => {
+//     it("decides if the number inside is divisible by 3.", () => {
+//         expect(divideBy3(object1.number)).toEqual("15 is divisible by three")
+//         expect(divideBy3(object2.number)).toEqual("0 is divisible by three")
+//         expect(divideBy3(object3.number)).toEqual("-7 is not divisible by three")
```
Comment (chrisyhlee) [inline code-challenges.test.js]: The prompt tells you that your function should take an object as an argument, not the number in that object.

File: code-challenges.test.js [hunk 2]
```diff
@@ -1,55 +1,111 @@
...
-// Pseudo code:
+// // Pseudo code: 
+
+//  1.) Write a function that take in the array and add the map HOF to iterate in the array. 
+
+//  2.) In the map HOF write a function that takes the value you are iterating and split the strings into individual strings of letters.
+
+//  3.) add another map HOF to iterate each individual letter and add a conditional statement to return the 0 index in the string uppercased and the rest of the string as is. 
+
+//  4.) Add the .join('') with single quotations to join the string "letters" back to string "words".
```
Comment (chrisyhlee) [inline code-challenges.test.js:88]: Good logic here!

**Review by chrisyhlee — CHANGES_REQUESTED** [review]
Hey Paul, I saw your note about challenge 3, hope everything is okay now! I can't approve this PR until there is a valid attempt, so once you complete challenge 3, you can add, commit, and push up to the same branch.

**Comment by Phalco1** [comment]
I’ll do it asap

**Comment by Phalco1** [comment]
I’ll do it during lunch

File: code-challenges.test.js [hunk 3]
```diff
@@ -64,4 +120,33 @@ const vowelTester3 = "challenges"
...
+
+// 3.) Create a conditional "if" statement that states: if any values in the array are identified, the .includes() method will return the value that is true(vowels).
+
+// 4.) Return the index as is, which will be the index where the value(vowels) in the array is identified.
+
+
+const vowels =["a", "e", "i", "o", "u"]
+
+const firstVowel = (string) => {
+    let theVowel = []
```
Comment (chrisyhlee) [inline code-challenges.test.js:139]: no need for this variable!

**Review by chrisyhlee — APPROVED** [review]
Thanks for making the changes, and good work on this assessment! I left comments throughout for you to look at, if you have any further questions please let me know! I suggest you focus on elaborating more in your interview questions in order to fully convey the extent of your knowledge to the interviewer. Your pseudo code is doing a good job in covering each step, and you have a solid basic understanding of these concepts so far. This assessment currently displays Novice, but you are very close to Advanced Beginner, solid work!

**Comment by Phalco1** [comment]
Thanks a lot for the feedback.

**Pull request merged**